mac builds failing on Path is a directory: rm returned EISDIR

<!-- Which version of electron-builder are you using? -->
<!-- Please always try to use latest version before reporting any issue. -->
* **Version**:  22.11.5

<!-- Which version of electron are you using? -->
<!-- Please verify if the error is electron related before reporting any issues. -->
* **Electron Version**: 12.0.9
* **Electron Type (current, beta, nightly)**: current
<!-- Which version of electron-updater are you using (if applicable)? -->

<!-- For which target are you building for? -->
* **Target**: mac

<!-- Enter your issue details below this comment. -->
<!-- If you want, you can donate to increase issue priority (https://www.electron.build/donate) -->

With the bump from electron-builder 22.11.3 to 22.11.4 (and 22.11.5) I started getting failures building my simple Electron app on macOS. Rolling back to 22.11.3 and keeping all other deps current fixed the issue for us. 

Here's the error from our github action after dependabot opened a PR to bump back to the latest electron-builder:

```
> npm run compile && electron-builder --mac --publish never


> chef-workstation-app@0.2.3 compile /Users/runner/work/chef-workstation-app/chef-workstation-app
> tsc && copyfiles -u 1 "src/**/*.html" build

  • electron-builder  version=22.11.5 os=19.6.0
  • loaded configuration  file=/Users/runner/work/chef-workstation-app/chef-workstation-app/electron-builder.yml
  • packaging       platform=darwin arch=x64 electron=12.0.9 appOutDir=dist/mac
  • downloading     url=https://github.com/electron/electron/releases/download/v12.0.9/electron-v12.0.9-darwin-x64.zip size=80 MB parts=6
  • downloaded      url=https://github.com/electron/electron/releases/download/v12.0.9/electron-v12.0.9-darwin-x64.zip duration=743ms
  ⨯ Path is a directory: rm returned EISDIR (is a directory) /Users/runner/work/chef-workstation-app/chef-workstation-app/dist/mac/Chef Workstation App.app/Contents/Resources/am.lproj  failedTask=build stackTrace=SystemError [ERR_FS_EISDIR]: Path is a directory: rm returned EISDIR (is a directory) /Users/runner/work/chef-workstation-app/chef-workstation-app/dist/mac/Chef Workstation App.app/Contents/Resources/am.lproj
    at internal/fs/utils.js:692:23
    at FSReqCallback.oncomplete (fs.js:193:5)
From previous event:
    at processImmediate (internal/timers.js:462:21)
npm ERR! code ELIFECYCLE
From previous event:
npm ERR! errno 1
    at beforeCopyExtraFiles (/Users/runner/work/chef-workstation-app/chef-workstation-app/node_modules/app-builder-lib/src/electron/ElectronFramework.ts:96:27)
npm ERR! chef-workstation-app@0.2.3 build-mac: `npm run compile && electron-builder --mac --publish never`
    at MacPackager.doPack (/Users/runner/work/chef-workstation-app/chef-workstation-app/node_modules/app-builder-lib/src/platformPackager.ts:271:7)
npm ERR! Exit status 1
    at MacPackager.pack (/Users/runner/work/chef-workstation-app/chef-workstation-app/node_modules/app-builder-lib/src/macPackager.ts:177:7)
npm ERR! 
    at Packager.doBuild (/Users/runner/work/chef-workstation-app/chef-workstation-app/node_modules/app-builder-lib/src/packager.ts:441:9)
    at Object.executeFinally (/Users/runner/work/chef-workstation-app/chef-workstation-app/node_modules/builder-util/src/promise.ts:12:14)
npm ERR! Failed at the chef-workstation-app@0.2.3 build-mac script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/runner/.npm/_logs/2021-05-24T21_57_38_218Z-debug.log
    at Packager._build (/Users/runner/work/chef-workstation-app/chef-workstation-app/node_modules/app-builder-lib/src/packager.ts:376:31)
    at Packager.build (/Users/runner/work/chef-workstation-app/chef-workstation-app/node_modules/app-builder-lib/src/packager.ts:337:12)
    at Object.executeFinally (/Users/runner/work/chef-workstation-app/chef-workstation-app/node_modules/builder-util/src/promise.ts:12:14)
```

That job is available here: https://github.com/chef/chef-workstation-app/pull/368/checks?check_run_id=2659855178


Thanks for everything. Electron-builder has been super helpful to use for the project.

--- a/packages/app-builder-lib/src/electron/ElectronFramework.ts
+++ b/packages/app-builder-lib/src/electron/ElectronFramework.ts
@@ -102,7 +102,7 @@ async function beforeCopyExtraFiles(options: BeforeCopyExtraFilesOptions) {
 
         const language = file.substring(0, file.length - langFileExt.length)
         if (!wantedLanguages.includes(language)) {
-          return fs.rm(path.join(resourcesDir, file), { force: true })
+          return fs.rm(path.join(resourcesDir, file), { recursive: true, force: true })
         }
         return
       },

